kong/plugins/http-log/sender.lua

```diff
@@ -73,18 +73,19 @@ local function send(self, bodies)
       },
       body = body,
     })
+    ok, err2 = httpc:set_keepalive(self.keepalive)
```

Comment:
So, this is the same issue: if you call `set_keepalive` before calling `receive` (what `read_body` does under the hood), you'll get an error because the Lua-land socket object will be flipped to the "closed" state.

I have updated my previous suggestion to reflect how we usually handle this scenario (keepalive on a failed request), plus updated the return statement which should only be called if `res == nil`.

It seems like a test is warranted here, since the CI failed to catch any issue with the last few proposals around this patch, which is worrying if we want to prevent future regressions...